packages/components/src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,145 @@
+import type { SystemProp, Theme } from "@xstyled/styled-components";
+import React from "react";
+import * as HeroOutlineIcons from "@heroicons/react/24/outline";
+import { Box } from "../../primitives/Box";
+import { Icon } from "../Icon";
+
+type IconNames = keyof typeof HeroOutlineIcons;
+
+type BadgeSizeOptions = "large" | "small";
+type BadgeVariantOptions =
+  | "active"
+  | "inactive"
+  | "primary"
+  | "success"
+  | "warning";
+
+export interface BadgeProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, "color"> {
+  /** The contents of the badge. Can be text. */
+  children: NonNullable<React.ReactNode>;
+  /** Sets the badge to be disabled. This is still being discussed with design whether we need the disabled prop or not. */
+  disabled?: boolean;
```

Comment:
I think @richbachman left a comment about the disabled prop, which is not solved yet.

Reply:
You probably need to talk with the design team again because I think it is on the Figma layout. It should be another variant, even though I'm not sure if `disabled` is a good name for it. We already have inactive, which is weird. Why do we need two different colors for the same thing? :thinking: